pkg/operator/encryption/shared_encryption.go

```diff
@@ -397,7 +405,15 @@ func keysToProviders(grKeys keysState) []apiserverconfigv1.ProviderConfiguration
 
 	providers := make([]apiserverconfigv1.ProviderConfiguration, 0, len(allKeys)+1) // one extra for identity
 
-	for _, key := range allKeys {
+	// having identity as a key is problematic because IdentityConfiguration cannot store any data.
+	// we need to be able to trace back to the secret so that it can move through the key state machine.
```

Comment:
do you primarily mean `pod_state_controller_encryption`? If that's the case I guess having special logic just for this case wouldn't work? (A code that ignores identity keys and doesn't trace them back)

Reply:
Right we are basically trying to partially round trip:

`encryptionState` -> `encryptionConfig` -> `all keys and secrets we care about` (subset of `encryptionState`) -> `encryptionConfig` -> ...